content/source/docs/enterprise/api/oauth-clients.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: enterprise2
+page_title: "OAuth Clients - API Docs - Terraform Enterprise"
+sidebar_current: "docs-enterprise2-api-oauth-clients"
+---
+
+# OAuth Clients API
+
+-> **Note**: These API endpoints are in beta and are subject to change.
+
+An OAuth Client represents the connection between an organization and a VCS provider.
+
+## Create an OAuth Client
+
+`POST /organizations/:organization_name/oauth-clients`
+
+Parameter            | Description
+---------------------|------------
+`:organization_name` | The name of the organization that will be connected to the VCS provider. The organization must already exist in the system, and the user must have permissions to initiate the connection.
+
+This endpoint allows you to create a VCS connection between an organization and a VCS provider (GitHub, Bitbucket, or GitLab) for use when creating or setting up workspaces. By using this API endpoint, you can provide a pre-generated OAuth token string instead of going through the process of creating a GitHub/GitLab OAuth Application or Bitbucket App Link. To learn how to generate one of these token strings for your VCS provider, you can read the following documentation:
+
+* GitHub and GitHub Enterprise: [https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
+* Bitbucket and Bitbucket Server: [https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html](https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html)
+* GitLab, GitLab Community Edition, and GitLab Enterprise Edition: [https://docs.gitlab.com/ce/user/profile/personal_access_tokens.html#creating-a-personal-access-token](https://docs.gitlab.com/ce/user/profile/personal_access_tokens.html#creating-a-personal-access-token)
+
+Result  | Status and response
+--------|------------------------
+Success | HTTP 201 and a JSON API document (`type: "oauth-clients"`).
+Failure | HTTP 4XX and a JSON API list of error messages.
```

Comment:
Well it's not reviewed and merged yet, but Dylan suggested some improvements to the status code table, and it's in a PR at https://github.com/hashicorp/terraform-website/pull/241. Buuuuut you can skip it since it's not approved. 😅